app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,78 @@ const validClusterAuth = async (me, queryName, context) => {
   }
 };
 
-var getAllowedChannels = async(me, org_id, action, field, context)=>{
+// get allowed resources
+const getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
```

Comment:
The variable naming like `searchByTags` suggests that it should be a boolean -- "search by tags, yes or no?".  It's also not obvious that searching by UUID is only allowed if searching groups.  This is because the function was made with special cases for how existing code actually queries different types, rather than making a really _generic_ function that can query any type by tags (any?) or by uuids (groups only?) or by a passed query (subscriptions only?)
```suggestion
/*
Get resources specified by `type`, restricted to those the user is authorized to `action` (which could be none, returning an empty array).
If querying Groups, can query groups matching a passed array of group UUIDs.
If querying Subscriptions, can query subscriptions matching a passed query object.
*/
const getAllowedResources = async(me, org_id, action, type, queryName, context, searchTags=null, searchUuids=null, searchQuery=null)=>{
```

Reply:
added